### ISSUE TITLE
net after quantize_qat can not improve accuracy

### ISSUE THREAD

**Issue by yongchaoding** [pytorch/pytorch#28071]
### Please note that this issue tracker is not a help form and this issue will be closed.
I have a vgg net, and want to using training-aware quant function in pytorch-1.3, then i do as follows:

```python
net.eval()
# Add quant config
net.qconfig = torch.quantization.get_default_qat_qconfig('fbgemm')
# merge conv+bn+relu
fuse_list = [['features.0', 'features.1', 'features.2'], \
             ['features.3', 'features.4', 'features.5'], \
             ['features.7', 'features.8', 'features.9'], \
             ['features.10', 'features.11', 'features.12'], \
             ['features.14', 'features.15', 'features.16'], \
             ['features.17', 'features.18', 'features.19'], \
             ['features.20', 'features.21', 'features.22'], \
             ['features.24', 'features.25', 'features.26'], \
             ['features.27', 'features.28', 'features.29'], \
             ['features.30', 'features.31', 'features.32'], \
             ['features.34', 'features.35', 'features.36'], \
             ['features.37', 'features.38', 'features.39'], \
             ['features.40', 'features.41', 'features.42']]
net = torch.quantization.fuse_modules(net, fuse_list)
# Add quant and dequant function
net = torch.quantization.QuantWrapper(net)
# quant and train model
net = torch.quantization.quantize_qat(net, train_echo, trainloader)
print(net)
```

in train_echo, the code is follows:

```python
def train_echo(net, data):
    net.train()
    optimizer = optim.SGD(net.parameters(), lr=args.lr, momentum=0.9, weight_decay=5e-4)
    for name, param in net.named_parameters():
        if param.requires_grad:
            print(name)
    for echo in range(0, 1):
        train_loss = 0
        correct = 0
        total = 0
        for batch_idx, (inputs, targets) in enumerate(data):
            print(net.state_dict()['module.features.40.weight'][0])
            inputs, targets = inputs.to(device), targets.to(device)
            optimizer.zero_grad()
            outputs = net(inputs)
            loss = criterion(outputs, targets)
            loss.backward()
            optimizer.step()
            #print(net.state_dict()['module.features.40.weight'][0])
            train_loss += loss.item()
            _, predicted = outputs.max(1)
            total += targets.size(0)
            correct += predicted.eq(targets).sum().item()

            progress_bar(batch_idx, len(trainloader), 'Loss: %.3f | Acc: %.3f%% (%d/%d)'
                % (train_loss/(batch_idx+1), 100.*correct/total, correct, total))
```

I can see that parameters in net is modified every echo, but the Acc is always 10% which means randn in CIFAR10.
So I am very confused. Is any idea?

We have a set of [listed resources available on the website](https://pytorch.org/resources). Our primary means of support is our discussion forum:

- [Discussion Forum](https://discuss.pytorch.org/)

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100

**Comment by jerryzh168** [comment]
Does it train correctly without qat? can you post the result of training with and without qat?

**Comment by yongchaoding** [comment]
I solve this problem, the net should set into train, net.train(), and do quantize_qat, so it works
